### PR TITLE
CLC-4744 - Course Provisioning: Post-Demo Fixes

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira.ets.berkeley.edu/jira/browse/CLC-4744

### CODE REVIEW

File: app/assets/javascripts/angular/factories/canvasCourseProvisionFactory.js [hunk 1]
```diff
@@ -70,6 +71,20 @@
       });
     };
 
+    /*
+     * Used to sort courses by title in fillCourseSites method
+     */
+    var compareCourseTitle = function(a,b) {
+      if (a['title'] < b['title']) {
```
Comment (christianv) [inline app/assets/javascripts/angular/factories/canvasCourseProvisionFactory.js]: @redconfetti can we use `a.title` instead?

File: app/assets/javascripts/angular/factories/canvasCourseProvisionFactory.js [hunk 2]
```diff
@@ -70,6 +71,20 @@
       });
     };
 
+    /*
+     * Used to sort courses by title in fillCourseSites method
+     */
+    var compareCourseTitle = function(a,b) {
```
Comment (christianv) [inline app/assets/javascripts/angular/factories/canvasCourseProvisionFactory.js]: @redconfetti space after `,`

File: app/assets/templates/canvas_embedded/canvas_course_sections_form.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
   <ul class="cc-template-canvas-manage-sections-margin">
 
     <li data-ng-repeat="course in currentCourses" class="cc-template-canvas-manage-sections-form-course">
-      <div data-ng-click="course.collapsed = !course.collapsed">
+      <div data-ng-click="course.collapsed = !course.collapsed" aria-controls="{{course.course_id}}">
```
Comment (christianv) [inline app/assets/templates/canvas_embedded/canvas_course_sections_form.html]: @redconfetti can we use `ng-attr-...`? `ng-attr-aria-controls`
Definitely check whether everything still works after that change

File: app/assets/templates/canvas_embedded/canvas_course_sections_form.html [hunk 4]
```diff
@@ -15,7 +15,7 @@
         </span>
       </div>
 
-      <div class="cc-template-canvas-manage-sections-form-collapsible-container" data-ng-if="!course.collapsed">
+      <div id="{{course.course_id}}" class="cc-template-canvas-manage-sections-form-collapsible-container" data-ng-if="!course.collapsed" aria-expanded="{{!course.collapsed}}" role="region">
```
Comment (christianv) [inline app/assets/templates/canvas_embedded/canvas_course_sections_form.html]: @redconfetti `ng-attr`

File: app/assets/templates/canvas_embedded/canvas_course_sections_form.html [hunk 5]
```diff
@@ -38,14 +38,24 @@
                     <strong data-ng-bind="section.courseCode"></strong>
                   </label>
                 </div>
-                <div class="cc-template-canvas-manage-sections-section-label" data-ng-bind="section.section_label"></div>
-                <div class="cc-template-canvas-manage-sections-section-ccn" data-ng-bind="section.ccn"></div>
+
+                <div class="cc-template-canvas-manage-sections-section-label" data-ng-bind="section.section_label">
+                  <span data-ng-if="section.section_label" data-ng-bind="section.section_label"></span>
+                  <span data-ng-if="!section.section_label"></span>
```
Comment (christianv) [inline app/assets/templates/canvas_embedded/canvas_course_sections_form.html]: @redconfetti why do we still need these 2 `spans`? Can we also remove the `data-ng-if` here?

File: app/assets/templates/canvas_embedded/canvas_course_sections_form.html [hunk 6]
```diff
@@ -38,14 +38,24 @@
...
+                </div>
+
+                <div class="cc-template-canvas-manage-sections-section-ccn">
+                  <span data-ng-bind="section.ccn"></span>
```
Comment (christianv) [inline app/assets/templates/canvas_embedded/canvas_course_sections_form.html]: @redconfetti we can probably remove the `span` and do the `data-ng-bind` on the `div` element instead.

**Comment by raydavis** [comment]
@christianv , this looks good to go -- you concur?

**Comment by raydavis** [comment]
Whoops, sorry, @christianv -- missed your latest comments!

**Comment by redconfetti** [comment]
@christianv 
little-girl-merge-request

**Pull request merged**